### PR TITLE
complement for RL e2e test

### PR DESCRIPTION
**Pull request by shawnh2**
**What type of PR is this?**

remove the two TODO lines in RL e2e test.

**What this PR does / why we need it**:

- as for first TODO, wait for deployment/envoy-gateway to finish restart, so the deployment/envoy-ratelimit can be created, and the `Sleep` can be removed
- as for second TODO, make the resp and req more explicit

### CODE REVIEW

**Comment by shawnh2** [comment]
it's wired that the request to gateway may `connect: connection refused` few times and then got `200`

File: test/e2e/tests/ratelimit.go [hunk 1]
```diff
@@ -33,20 +33,48 @@ var RateLimitTest = suite.ConformanceTest{
...
+			}
+			lastFourthReq := http.MakeRequest(t, &lastFourthExpResp, gwAddr, "HTTP", "http")
+			if err := GotNTimesExpectedResponse(t, 1, suite.RoundTripper, lastFourthReq, lastFourthExpResp); err != nil {
+				t.Errorf("fail to get expected response at last fourth request: %v", err)
+			}
 		})
 	},
 }
+
+func GotNTimesExpectedResponse(t *testing.T, n int, r roundtripper.RoundTripper, req roundtripper.Request, resp http.ExpectedResponse) error {
```
Comment (arkodg) [inline test/e2e/tests/ratelimit.go:68]: instead of doing an exact check of `n` requests, you might need to use some until/eventual  logic and wait for some timeout duration until all `n` requests ended up generating the expected response / status code.
because the e2e test might start sending traffic before the route was configured, etc

File: tools/make/kube.mk [hunk 2]
```diff
@@ -102,12 +102,15 @@ install-ratelimit:
 	@$(LOG_TARGET)
 	kubectl apply -f examples/redis/redis.yaml
 	kubectl rollout restart deployment envoy-gateway -n envoy-gateway-system
+	kubectl rollout status --watch --timeout=5m -n envoy-gateway-system deployment/envoy-gateway
```
Comment (arkodg) [inline tools/make/kube.mk:105]: why is this needed ?

Comment (arkodg) [reply tools/make/kube.mk:105]: the next line checks for `condition=Available`

Comment (shawnh2) [reply tools/make/kube.mk:105]: because the `rollout restart` does not affect the condition of deployments.

even through its replicaset is not ready yet, the condition of deployments is still `Available`. so the `rollout status` here watches till its replicaset gets ready.

Comment (arkodg) [reply tools/make/kube.mk:105]: thanks for the clarification, seems weird that it doesnt reset the status

File: test/e2e/tests/ratelimit.go (hunk 1 above)
Comment (shawnh2) [reply test/e2e/tests/ratelimit.go]: thanks, right now i wait till get the first 200, and then fire the rest three requests

Comment (arkodg) [reply test/e2e/tests/ratelimit.go]: thanks, re-read the logic and it looks good !

**Review by arkodg — APPROVED** [review]

**Review by zirain — APPROVED** [review]
